docs/readthedocs/source/doc/UserGuide/databricks.md

```diff
@@ -8,73 +8,91 @@ You can run BigDL program on the [Databricks](https://databricks.com/) cluster a
 
-![](images/upload-success.png)
+```bash
+#!/bin/bash
 
-Right click the uploaded zip package and choose **Copy path**, copy the **Spark API Format** path.
+# install bigdl-orca, add other bigdl modules if you need
+/databricks/python/bin/pip install pip install --pre --upgrade bigdl-orca-spark3
 
-![](images/copy-path.png)
+# install other necessary libraries, here we install libraries needed in this tutorial
+/databricks/python/bin/pip install ray[default]
+/databricks/python/bin/pip install tensorflow==2.9.1
+/databricks/python/bin/pip install pyarrow==8.0.0
+/databricks/python/bin/pip install psutil
 
-**Then install the zip package from DBFS.** In the left panel, click **Compute > choose your cluster > Libraries > Install new > Library Source(DBFS/ADLS) > Library Type(Python Egg) > paste the path > Install**
+# copy bigdl jars to databricks
+cp /databricks/python/lib/python3.8/site-packages/bigdl/share/*/lib/*.jar /databricks/jars
+```
 
-![](images/install-zip.png)
+Then upload **init.sh** to DBFS.  In Databricks left panel, click **Data > DBFS > Choose or create upload directory > Right click > Upload here**.
 
-#### 4.2 Change the File Extension Name
+![](images/upload-init-script.png)
 
-You can simply change the **bigdl-spark_3.1.2-2.1.0-SNAPSHOT-python-api.zip** extension name(**.zip**) to **.egg**, since Egg is essentially a zip format package. Then in the left panel, click **Compute > choose your cluster > Libraries > Install new > Library Source(Upload) > Library Type(Python Egg) > Install**
+Now the init script is in DBFS, right click the init.sh and choose **Copy path**, copy the **Spark API Format** path.
 
-![](images/egg.png)
+![](images/copy-script-path.png)
 
-### **5. Set Spark configuration**
+### 3. Set Spark configuration
 
 On the cluster configuration page, click the **Advanced Options** toggle. Click the **Spark** tab. You can provide custom [Spark configuration properties](https://spark.apache.org/docs/latest/configuration.html) in a cluster configuration. Please set it according to your cluster resource and program needs.
 
 ![](images/Databricks5.PNG)
 
-See below for an example of Spark config setting needed by BigDL. Here it sets 2 core per executor. Note that "spark.cores.max" needs to be properly set below.
+See below for an example of Spark config setting **needed** by BigDL. Here it sets 2 core per executor. Note that "spark.cores.max" needs to be properly set below.
 
 ```
 spark.executor.cores 2
 spark.cores.max 4
 ```
 
-### **6. Run BigDL on Databricks**
+### 4. Install BigDL Libraries
+
+Use init script from [step 2](#2-generate-initialization-script) to install BigDL libraries. In the left panel, click **Compute > Choose your cluster > edit > Advanced options > Init Scripts > Paste init script path > Add > Restart cluster**.
+
+![](images/config-init-script.png)
+
+After restart the cluster, the needed libraries is all installed.
```

Comment:
restart -> restarting
the libraries specified in the init script are all installed